### PR TITLE
Upgrade core api dependency to support protocol update

### PR DESCRIPTION
**Pull request by PawelPawelec-RDX**

### CODE REVIEW

File: src/RadixDlt.NetworkGateway.PostgresIntegration/CommonDbContext.cs [hunk 1]
```diff
@@ -210,7 +210,8 @@ private static void HookupTransactions(ModelBuilder modelBuilder)
             .HasDiscriminator<LedgerTransactionType>(DiscriminatorColumnName)
             .HasValue<UserLedgerTransaction>(LedgerTransactionType.User)
             .HasValue<RoundUpdateLedgerTransaction>(LedgerTransactionType.RoundUpdate)
-            .HasValue<GenesisLedgerTransaction>(LedgerTransactionType.Genesis);
+            .HasValue<GenesisLedgerTransaction>(LedgerTransactionType.Genesis)
+            .HasValue<GenesisLedgerTransaction>(LedgerTransactionType.Flash);
```
Comment (krzlabrdx) [inline src/RadixDlt.NetworkGateway.PostgresIntegration/CommonDbContext.cs:214]: Invalid generic type of `HasValue`?

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Migrations/20240110094622_AddFlashTransactionType.cs [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+namespace RadixDlt.NetworkGateway.PostgresIntegration.Migrations
+{
+    /// <inheritdoc />
+    public partial class AddFlashTransactionType : Migration
+    {
+        /// <inheritdoc />
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.AlterDatabase()
```
Comment (krzlabrdx) [inline src/RadixDlt.NetworkGateway.PostgresIntegration/Migrations/20240110094622_AddFlashTransactionType.cs:77]: Seems that autogenerated migration went retard. Maybe we should write it manually?

**Review by krzlabrdx — CHANGES_REQUESTED** [review]

File: src/RadixDlt.NetworkGateway.PostgresIntegration/CommonDbContext.cs (hunk 1 above)
Comment (PawelPawelec-RDX) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/CommonDbContext.cs]: fixed.

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Migrations/20240110094622_AddFlashTransactionType.cs (hunk 2 above)
Comment (PawelPawelec-RDX) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/Migrations/20240110094622_AddFlashTransactionType.cs:77]: fixed. Tested and that doesn't throw when we only execute migrations but run on old code in api/aggregator.

**Review by krzlabrdx — APPROVED** [review]

**Pull request merged**